### ISSUE TITLE
The load API is blocked when the model fails to load

### ISSUE THREAD

**Issue by so2bin** [triton-inference-server/server#6536]
**Description**
Dynamic loading of a model using the load API is blocked when the model fails to load.

* The server logs:
![image](https://github.com/triton-inference-server/server/assets/9431914/0f092b08-c335-4d9c-86e9-7518f7f607d3)

* The client command and logs:
![image](https://github.com/triton-inference-server/server/assets/9431914/1a6cec0f-6201-4780-a470-bafb498b9eca)

**Triton Information**
The newest official tritonserver docker image: `tritonserver:23.10-pyt-python-py3`

**To Reproduce**
* Start the triton server with EXPLICIT mode:
```
tritonserver --model-repository=/home/aaa/triton/demo/ --log-verbose=1 --model-control-mode=explicit --load-model=*
```
* Add the "resnet" model folder to model repository: `/home/aaa/triton/demo/`
* The config.pbtxt of resnet model is:
```
name: "resnet"
backend: "python"
default_model_filename: "model.py"
```
* Write a wrong model config in `model.py` with wrong input number like this:
```python
import os
import torch
import time
from torchvision import models
import triton_python_backend_utils as pb_utils


def load_with_local_model():
    res50 = models.resnet50(pretrained=False)
    weights = torch.load(os.path.join(os.path.dirname(__file__), "resnet50.pth"))
    res50.load_state_dict(weights)
    return res50


class TritonPythonModel:
    @staticmethod
    def auto_complete_config(auto_complete_model_config):
        # here we got a wrong input config for resnet50.pth
        inputs = [{
            'name': 'INPUT0',
            'data_type': 'TYPE_FP32',
            'dims': [4],
            # this parameter will set `INPUT0 as an optional input`
            'optional': True
        }, {
            'name': 'INPUT1',
            'data_type': 'TYPE_FP32',
            'dims': [4]
        }]
        outputs = [{
            'name': 'OUTPUT0',
            'data_type': 'TYPE_FP32',
            'dims': [4]
        }, {
            'name': 'OUTPUT1',
            'data_type': 'TYPE_FP32',
            'dims': [4]
        }]
        config = auto_complete_model_config.as_dict()
        input_names = []
        output_names = []
        for input in config['input']:
            input_names.append(input['name'])
        for output in config['output']:
            output_names.append(output['name'])

        for input in inputs:
            if input['name'] not in input_names:
                auto_complete_model_config.add_input(input)
        for output in outputs:
            if output['name'] not in output_names:
                auto_complete_model_config.add_output(output)

        auto_complete_model_config.set_max_batch_size(0)

        return auto_complete_model_config

    def initialize(self, args):
        self.model = load_with_local_model()

    def execute(self, requests):
        responses = []
        for request in requests:
            time.sleep(0.1)
            print("----------->>> infer with request")
        return responses

    def finalize(self):
        print('Cleaning up...')

```
* Call load API command: 
```
curl -v -XPOST -d '{}' http://127.0.0.1:8000/v2/repository/models/resnet/load
```
* Now, we can got a pending request with no reponse.

**Expected behavior**
The loading API should reponse with error after the model loading is failed.

**Comment by krishung5** [comment]
Hi @so2bin, I follow the steps you provided and load your model, and I'm getting the response with error message on the client side.
```
root@463668eb3d00:/opt/tritonserver# curl -v -XPOST -d '{}' http://127.0.0.1:8000/v2/repository/models/resnet/load
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 127.0.0.1:8000...
* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
> POST /v2/repository/models/addsub/load HTTP/1.1
> Host: 127.0.0.1:8000
> User-Agent: curl/7.81.0
> Accept: */*
> Content-Length: 2
> Content-Type: application/x-www-form-urlencoded
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Content-Type: application/json
< Content-Length: 389
< 
* Connection #0 to host 127.0.0.1 left intact
{"error":"load failed for model 'addsub': version 1 is at UNAVAILABLE state: Internal: ValueError: input 'INPUT0' in auto-complete-config function for model 'addsub' contains property other than 'name', 'data_type' and 'dims'.\n\nAt:\n  /opt/tritonserver/backends/python/triton_python_backend_utils.py(392): add_input\n  /workspace/models/addsub/1/model.py(49): auto_complete_config\n;\n"}
```
Are you calling the load API within the same container or on a different container?

Besides, this error should be fixed in Triton 23.11 release by this [PR](https://github.com/triton-inference-server/python_backend/commit/0f1221129a01b067b93a4abc4a2c30a9e2856e01). The optional field should be acceptable for input.

**Comment by krishung5** [comment]
Closing issue due to lack of activity. Please re-open the issue if you would like to follow up with this issue.
